src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -451,9 +464,14 @@ def _clean_parent(self):
                 _("A parent product can't have stockrecords."))
 
     def save(self, *args, **kwargs):
+        is_product_has_options = len(self.options) > 0 if self.pk else False
```

Comment:
I think this will cause `has_options` to be incorrect when creating a new product whose product class has options.